src/baggage/remote_baggage_restriction_manager.js

```diff
@@ -0,0 +1,161 @@
+        this._refreshInterval = options.refreshInterval || DEFAULT_REFRESH_INTERVAL;
+        this._host = options.host || DEFAULT_BAGGAGE_HOST;
+        this._port = options.port || DEFAULT_BAGGAGE_PORT;
+        this._failClosed = options.failClosed || DEFAULT_FAIL_CLOSED;
+        this._restrictionsMap = {};
+        this._initialized = false;
+        this._onRestrictionsUpdate = options.onRestrictionsUpdate;
+
+        this._refreshBaggageRestrictions();
+
+        if (options.refreshInterval !== 0) {
+            let randomDelay: number = Math.random() * this._refreshInterval;
+            this._initialDelayTimeoutHandle = setTimeout(this._afterInitialDelay.bind(this), randomDelay);
+        }
+    }
+
+    isValidBaggageKey(key: string): [boolean, number] {
+        if (!this._initialized) {
+            if (this._failClosed) {
+                return [false, 0];
+            }
+            return [true, constants.DEFAULT_MAX_VALUE_LENGTH];
+        }
+        if (key in this._restrictionsMap) {
+            return [true, this._restrictionsMap[key]]
+        }
+        return [false, 0];
+    }
+
+    _afterInitialDelay(): void {
+        this._refreshIntervalHandle = setInterval(
+            this._refreshBaggageRestrictions.bind(this),
+            this._refreshInterval
+        );
+    }
+
+    _refreshBaggageRestrictions() {
+        let serviceName: string = encodeURIComponent(this._serviceName);
+        let success: Function = (body) => {
+            this._parseBaggageRestrictionServerResponse(body);
+        };
+        let error: Function = (err) => {
+            this._logger.error(`Error in fetching baggage restrictions: ${err}.`);
+            this._metrics.baggageRestrictionsUpdateFailure.increment(1);
+        };
+        Utils.httpGet(this._host, this._port, `/baggageRestrictions?service=${serviceName}`, success, error);
+    }
+
+    _parseBaggageRestrictionServerResponse(body: string) {
+        let restrictions;
+        try {
+            restrictions = JSON.parse(body);
+            if (!restrictions) {
+                throw 'Malformed response: ' + body;
+            }
+        } catch (error) {
+            this._logger.error(`Error in parsing baggage restrictions: ${error}.`);
+            this._metrics.baggageRestrictionsUpdateFailure.increment(1);
+            return;
+        }
+        try {
+            this._updateRestrictions(restrictions);
+            this._initialized = true;
+            this._metrics.baggageRestrictionsUpdateSuccess.increment(1);
+        } catch (error) {
+            this._logger.error(`Error in updating baggage restrictions: ${error}.`);
+            this._metrics.baggageRestrictionsUpdateFailure.increment(1);
+            return;
+        }
+        if (this._onRestrictionsUpdate) {
+            this._onRestrictionsUpdate(this);
+        }
+    }
+
+    _updateRestrictions(restrictions: Array<BaggageRestriction>) {
+        let restrictionsMap: { [key: string]: number } = {};
```

Comment:
sheeeeet, yeah this is a good idea... My only gripe with this approach is that in the idl, I defined it as: https://github.com/uber/jaeger-idl/blob/master/thrift/baggage.thrift which I prefer over map<string, i32> because it's harder to tell what the variables mean without having a comment describing it.

Now the debate becomes lets leave the thrift idl as is but have the agent convert the array into a map before returning it. This does make sense but again you have that issue where as the consumer of the json, you dont know what the variables in the map symbolizes. By returning it as json array instead of map, each variable is keyed correctly so you know what they symbolize. wdyt?